src/python/pants/help/help_formatter.py

```diff
@@ -50,6 +57,8 @@ def add_option(ohis, *, category=None):
         lines.append(self._maybe_green(f'{display_scope}:'))
         if description:
           lines.append(description)
+        if related_subsystem_scopes:
+          lines.append(f"Related subsystems: {', '.join(sorted(related_subsystem_scopes))}")
```

Comment:
Suggestions for better wording appreciated! I'm not sure if `Related subsystems` is the right thing to call this? Maybe `Related option scopes`? 

Also, we should possibly say something like `Run `./pants help` on these`? But then the wording gets pretty verbose.